### PR TITLE
[10.0][ADD] account_check_report

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Add account_check_report module to v10.

This module is intended to print a report of invoice payments done through a check.
image

cc ~ @aheficent

### CODE REVIEW

**Review by AaronHForgeFlow — COMMENTED** [review]
I seems the check amount is overlapped when having many lines. Can you check @ageficent ?

image

**Comment by AdriaGForgeFlow** [comment]
> I seems the check amount is overlapped when having many lines. Can you check @ageficent ?

Can you check now? Thanks!

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: It works like a charm! Thanks

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM :+1:

**Comment by AaronHForgeFlow** [comment]
Please do not merge this until this question is answered: https://github.com/Eficent/account-payment/pull/2

**Review by AaronHForgeFlow — APPROVED** [review]

**Review by AdriaGForgeFlow — COMMENTED** [review]
Last commits LGTM :+1:

File: account_check_report/report/account_check_report.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+     <report
+        id="action_report_check_report"
+        model="account.payment"
+        string="Payment Check Report"
+        report_type="qweb-pdf"
+        name="account_check_report.check_report"
+        file="account_check_report.check_report"/>
+
+</odoo>
```
Comment (max3903) [inline account_check_report/report/account_check_report.xml]:
```suggestion
</odoo>
```

Comment (max3903) [inline account_check_report/report/account_check_report.xml:74]: Missing newline.

Otherwise :+1:

**Review by max3903 — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
> Please do not merge this until this question is answered: [Eficent#2](https://github.com/Eficent/account-payment/pull/2)

After the refactoring, please dismiss the above quoted comment

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
Hi @ageficent , it seems the refunded bills are excluded in the _get_paid_lines method. Can you check?

Thank you!

**Comment by AdriaGForgeFlow** [comment]
> Hi @ageficent , it seems the refunded bills are excluded in the _get_paid_lines method. Can you check?
> 
> Thank you!

Could you clarify a little bit the steps I need to follow to reproduce the error? I don't quite understand the issue.
Thanks!

**Comment by AaronHForgeFlow** [comment]
> Could you clarify a little bit the steps I need to follow to reproduce the error? I don't quite understand the issue.
> Thanks!

Sure: Create a vendor bill and a vendor refund for the same supplier. Then create a check payment for that supplier. Then reconcile the payment with the invoice and the refund. Then, in this report I should see both, the vendor bill and the refund. 

It seems there are some cases were the refund is not showing, but I am not completely sure about the reason. I will also try to take a look at this tomorrow.

**Comment by JordiBForgeFlow** [comment]
@ageficent @aheficent the latest commit fixes this problem. The way to solve it is rather tricky. See my comment in the code. Do you understand it?

image

**Review by AdriaGForgeFlow — COMMENTED** [review]
> @ageficent @aheficent the latest commit fixes this problem. The way to solve it is rather tricky. See my comment in the code. Do you understand it?

Understood :+1: Thanks!

**Comment by AaronHForgeFlow** [comment]
@jbeficent  I have done a functional review to this module and seems to work ok now. Did you find a case where the report does not work?

**Comment by AaronHForgeFlow** [comment]
It seems last commit 68383015f89132816347e7a8ce82fbeb72d6d1ca does the correct reconciliation with refunds.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Could not find anything wrong in this PR

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Comment by AaronHForgeFlow** [comment]
Ready to merge

**Pull request merged**